### ISSUE TITLE
Fails building RPI_PICO2

### ISSUE THREAD

**Issue by peterhinch** [mattytrentini/mpbuild#55]
After doing an `mprbuild clean` I issued
```bash
$ mpbuild build RPI_PICO2
```
This was the outcome
```bash
[lines omitted]
[ 37%] Building C object CMakeFiles/firmware.dir/home/adminpete/micropython/extmod/modlwip.c.obj
[ 37%] Building C object CMakeFiles/firmware.dir/home/adminpete/micropython/extmod/modmachine.c.obj
In file included from /home/adminpete/micropython/ports/rp2/machine_pwm.c:32,
                 from /home/adminpete/micropython/extmod/machine_pwm.c:50:
/home/adminpete/micropython/lib/pico-sdk/src/rp2_common/hardware_pwm/include/hardware/pwm.h: In function 'pwm_set_irq_mask_enabled':
/home/adminpete/micropython/lib/pico-sdk/src/rp2_common/hardware_pwm/include/hardware/pwm.h:658:55: error: expected ',' before ')' token
     static_assert(PWM_IRQ_WRAP_1 == PWM_IRQ_WRAP_0 + 1);
                                                       ^
                                                       ,
[ 37%] Building C object CMakeFiles/firmware.dir/home/adminpete/micropython/extmod/modonewire.c.obj
[ 38%] Building C object CMakeFiles/firmware.dir/home/adminpete/micropython/extmod/modnetwork.c.obj
make[3]: *** [CMakeFiles/firmware.dir/build.make:2100: CMakeFiles/firmware.dir/home/adminpete/micropython/extmod/machine_pwm.c.obj] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/adminpete/micropython/ports/rp2/build-RPI_PICO2'
make[2]: *** [CMakeFiles/Makefile2:2184: CMakeFiles/firmware.dir/all] Error 2
make[2]: Leaving directory '/home/adminpete/micropython/ports/rp2/build-RPI_PICO2'
make[1]: *** [Makefile:103: all] Error 2
make[1]: Leaving directory '/home/adminpete/micropython/ports/rp2/build-RPI_PICO2'
-e See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [Makefile:59: all] Error 1
make: Leaving directory '/home/adminpete/micropython/ports/rp2'
╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                              Flashing via UF2 bootloader                                                               │
│                                                                                                                                                        │
│ To get the board in bootloader mode ready for the firmware update, execute machine.bootloader() at the MicroPython REPL.  Alternatively, hold down the │
│ BOOTSEL button while plugging the board into USB.  The uf2 file below should then be copied to the USB mass storage device that appears.  Once         │
│ programming of the new firmware is complete the device will automatically reset and be ready for use.                                                  │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```
There would seem to be something amiss in the SDK.

**Comment by mattytrentini** [comment]
Yes, it need a newer version of GCC (or a newer version of the pico_sdk, [where it's been fixed](https://github.com/raspberrypi/pico-sdk/issues/1788) and no longer needs a newer compiler). 

You can workaround the issue with the following, which uses a newer GCC version:

```bash
mpbuild build RPI_PICO2 --build-container micropython/build-micropython-arm:bookworm
```

The goal is to switch entirely to use that container for all ARM builds (once we have confidence there are no regressions) and/or allowing more granular container config per board/variant. #54 also has a good interim fix for this, as well as using the right container for the `RISCV` variant.

There's a little more detail in #27.

**Comment by mattytrentini** [comment]
Resolved with v0.6.
